### ISSUE TITLE
{"errors":[{"status":"422","title":"found unpermitted parameter: :volume"}]}

### ISSUE THREAD

**Issue by pdurbin** [CDLUC3/counter-processor#8]
@sfisher you're going to hate me but I didn't test pull request #5 with the DataCite hub. Now I'm getting "found unpermitted parameter: :volume" when I try to send my report.

Here's how the error looks:

```
{
  "errors": [
    {
      "status": "422",
      "title": "found unpermitted parameter: :volume"
    }
  ]
}
```

Here's the command I'm using `curl --header "Content-Type: application/json; Accept: application/json" -H "Authorization: Bearer $JSON_WEB_TOKEN" -X POST https://api.test.datacite.org/reports/ -d @src/test/java/edu/harvard/iq/dataverse/makedatacount/sushi_sample_logs3.json`

That sushi_sample_logs3.json report can be found at https://github.com/IQSS/dataverse/commit/36fa81a

It looks like you already reported this at https://github.com/datacite/sashimi/issues/23

We should revert my pull request, right? Or add a boolean so we can turn on and off the sending of volume?

**Comment by sfisher** [comment]
Hi there.

I made some changes and a push to the repository.  This is also in the non-daily-logs branch.  See if that does what you need and let me know.

You'll need to set "output_volume: false" in the yml config file and then if you want volume, set an environment variable for OUTPUT_VOLUME to true.  (or the opposite way around.)

For example:

```
YEAR_MONTH=2019-01 UPLOAD_TO_HUB=False OUTPUT_VOLUME=true ./main.py
```

For a while I had a flag for "only output stats" or something like that and that wouldn't suck in any log files at all and would just work with whatever was already in the sqlite database.  Looks like I probably should bring it back.

It actually always calculates volume, but it just wasn't outputting it.

**Comment by pdurbin** [comment]
@sfisher I just looked at e550b77 and it seems to be exactly what we need. Thanks! Are you planning merging it to master and cutting a release?

**Comment by pdurbin** [comment]
@sfisher on second thought, while the `OUTPUT_VOLUME` boolean in e550b77 is helpful (and exactly what I asked for above!) what we really want is two versions of the JSON report. The idea is that the version without the volume will be sent to the DataCite hub until https://github.com/datacite/sashimi/issues/23 is resolved. The version with the volume would be taken into Dataverse but only after we get some resolution on the "country-volume" idea at https://github.com/datacite/sashimi/issues/23#issuecomment-459506293 (which might require a CoP change?!? see https://twitter.com/philipdurbin/status/1091344788568330240 ).

Bottom line, the boolean alone won't help us we'll probably run Counter Processor 0.0.1 so we can use it with the DataCite hub. So please don't feel any pressure from me to put out another release! Thanks!

**Comment by sfisher** [comment]
I'm going to close this since I think it's resolved and the `volume: False` flag I believe allows it up upload to the DataCite hub.
